### PR TITLE
Sd 955 missing analyses rest endpoint

### PR DESCRIPTION
**Pull request by jstucke**
- added REST endpoint for missing analyses and files

### CODE REVIEW

File: src/web_interface/rest/rest_base.py [hunk 1]
```diff
@@ -9,17 +9,19 @@
...
+        self.api.add_resource(RestBinarySearch, '/rest/binary_search', '/rest/binary_search/<search_id>', methods=['GET', 'POST'], resource_class_kwargs={'config': config})
         self.api.add_resource(RestCompare, '/rest/compare', '/rest/compare/<compare_id>', methods=['GET', 'PUT'], resource_class_kwargs={'config': config})
-        self.api.add_resource(RestFirmware, '/rest/firmware', '/rest/firmware/<uid>', methods=['GET', 'PUT'], resource_class_kwargs={'config': config})
         self.api.add_resource(RestFileObject, '/rest/file_object', '/rest/file_object/<uid>', methods=['GET'], resource_class_kwargs={'config': config})
-        self.api.add_resource(RestBinarySearch, '/rest/binary_search', '/rest/binary_search/<search_id>', methods=['GET', 'POST'], resource_class_kwargs={'config': config})
+        self.api.add_resource(RestFirmware, '/rest/firmware', '/rest/firmware/<uid>', methods=['GET', 'PUT'], resource_class_kwargs={'config': config})
+        self.api.add_resource(RestMissingAnalyses, '/rest/missing_analyses', methods=['GET'], resource_class_kwargs={'config': config})
```
Comment (dorpvom) [inline src/web_interface/rest/rest_base.py:24]: Hm. Maybe `/rest/missing` suffices. But if you want to keep analysis in you have to make it singular to adhere to REST best practices (i.e. `/rest/missing_analysis`).

File: src/web_interface/rest/rest_missing_analyses.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        self.config = kwargs.get('config', None)
+
+    def get(self):
+        with ConnectTo(FrontEndDbInterface, self.config) as db:
+            missing_files = self._make_json_serializable(db.find_missing_files())
+            missing_analyses = self._make_json_serializable(db.find_missing_analyses())
+        return success_message(dict(missing_files=missing_files, missing_analyses=missing_analyses), self.URL)
+
+    @staticmethod
+    def _make_json_serializable(set_dict: Dict[str, Set[str]]) -> Dict[str, List[str]]:
```
Comment (dorpvom) [inline src/web_interface/rest/rest_missing_analyses.py:23]: I'd like to get rid of this utilizing `common_helper_encoding.ReportEncoder` but that's an issue for another PR.

**Comment by dorpvom** [comment]
Can you add some simple unit tests?

**Comment by dorpvom** [comment]
Also this needs documentation in the REST API wiki

File: src/test/integration/web_interface/rest/test_rest_missing_analyses.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        test_fw_1 = create_test_firmware()
+        missing_uid = 'uid1234'
+        test_fw_1.files_included.add(missing_uid)
+        self.db_backend.add_firmware(test_fw_1)
+
+        response = json.loads(self.test_client.get('/rest/missing_analyses', follow_redirects=True).data.decode())
+        assert 'missing_files' in response
+        assert test_fw_1.uid in response['missing_files']
+        assert missing_uid in response['missing_files'][test_fw_1.uid]
+        assert response.get('missing_analyses') == {}
```
Comment (dorpvom) [inline src/test/integration/web_interface/rest/test_rest_missing_analyses.py:30]: Why the get?

File: src/test/integration/web_interface/rest/test_rest_missing_analyses.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        test_fo.virtual_file_path = {test_fw_1.uid: ['|foo|bar|']}
+        test_fw_1.processed_analysis['foobar'] = {'foo': 'bar'}
+        self.db_backend.add_firmware(test_fw_1)
+        self.db_backend.add_file_object(test_fo)
+
+        response = json.loads(self.test_client.get('/rest/missing_analyses', follow_redirects=True).data.decode())
+        assert 'missing_analyses' in response
+        assert test_fw_1.uid in response['missing_analyses']
+        assert test_fo.uid in response['missing_analyses'][test_fw_1.uid]
+        assert response.get('missing_files') == {}
```
Comment (dorpvom) [inline src/test/integration/web_interface/rest/test_rest_missing_analyses.py:45]: See above

File: src/test/integration/web_interface/rest/test_rest_missing_analyses.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        test_fw_1.files_included.add(missing_uid)
+        self.db_backend.add_firmware(test_fw_1)
+
+        response = json.loads(self.test_client.get('/rest/missing_analyses', follow_redirects=True).data.decode())
+        assert 'missing_files' in response
+        assert test_fw_1.uid in response['missing_files']
+        assert missing_uid in response['missing_files'][test_fw_1.uid]
+        assert response.get('missing_analyses') == {}
+
+    def test_rest_get_missing_analyses(self):
```
Comment (dorpvom) [inline src/test/integration/web_interface/rest/test_rest_missing_analyses.py:32]: Test is sort of confusing. But I don't see a better way given the task at hand.

File: src/web_interface/rest/rest_base.py (hunk 1 above)
Comment (jstucke) [reply src/web_interface/rest/rest_base.py:24]: changed to `/rest/missing`

File: src/test/integration/web_interface/rest/test_rest_missing_analyses.py (hunk 3 above)
Comment (jstucke) [reply src/test/integration/web_interface/rest/test_rest_missing_analyses.py:30]: changed

File: src/test/integration/web_interface/rest/test_rest_missing_analyses.py (hunk 4 above)
Comment (jstucke) [reply src/test/integration/web_interface/rest/test_rest_missing_analyses.py:45]: changed

File: src/test/integration/web_interface/rest/test_rest_missing_analyses.py (hunk 5 above)
Comment (jstucke) [reply src/test/integration/web_interface/rest/test_rest_missing_analyses.py:32]: added comment that explains how the test works

**Pull request merged**